.github/workflows/ci.yaml

```diff
@@ -0,0 +1,24 @@
+name: Java CI with Gradle
+
+on: [push]
+
+permissions:
+  contents: read
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up JDK 17
+        uses: actions/setup-java@v3
+        with:
+          java-version: '17'
+          distribution: 'temurin'
+      - name: Build with Gradle
+        uses: gradle/gradle-build-action@bd5760595778326ba7f1441bcf7e88b49de61a25
+        with:
+          arguments: build
+      - name: Test with Gradle
+        run: gradle clean test
```

Comment:
It's a good idea to separate building and testing the project.
`gradle build` operation compiles and tests the project. Then you run `gradle test` which tests it again. It's a commonly used operations locally but if we want to separate steps it does not suit us. 
You need to compile in one step, in a second test it. Try it out:
```
- name: Build 
 run: gradle classes
      
- name: Test
run: gradle test
```